### PR TITLE
fix(css website): Typography Fixes

### PR DESCRIPTION
**Pull request by DavidWeid**
## What does this PR do?
- Updates `hover:underline` class to use border-bottom
- Updates homepage community section links (2) for visual consistency
- Updates `li` behavior when containing elements such as `code` or `a` as a first or last child

## Pages to checkout in the preview:
Homepage
- https://deploy-preview-8860--vector-project.netlify.app/

`li` with links / code blocks
- https://deploy-preview-8860--vector-project.netlify.app/docs/setup/installation/platforms/docker/
- https://deploy-preview-8860--vector-project.netlify.app/highlights/2021-07-21-0-16-upgrade-guide/#vector_source_sink
- https://deploy-preview-8860--vector-project.netlify.app/docs/administration/validating/
- https://deploy-preview-8860--vector-project.netlify.app/docs/reference/cli/ (bullets at bottom)

### CODE REVIEW

**Comment by lucperkins** [comment]
@DavidWeid Love it! So glad you're going through and addressing some of these "broken window"-style issues. Could I perhaps trouble you with one more small thing I just noticed? In dark mode, the numbers in ordered lists come out looking a little dark, as you can see on a page [like this](https://vector.dev/docs/about/under-the-hood/guarantees/). No worries if you want to leave that for another PR but thought I'd point it out while you're at it.

**Review by lucperkins — APPROVED** [review]
Good stuff!

**Review by nsollecito — APPROVED** [review]

**Comment by DavidWeid** [comment]
@lucperkins Do you think it should be a different shade of gray from the text or does it not need to be different at all?

**Comment by lucperkins** [comment]
@DavidWeid Naw, I don't think it needs to be different from the text. Having the same color for the numbers/markers in `ol`s/`ul`s seems pretty standard to me.

File: website/cue/reference/cli.cue [hunk 1]
```diff
@@ -198,7 +198,7 @@ cli: {
 
 				Example:
 
-				```
+				```shell
```
Comment (lucperkins) [inline website/cue/reference/cli.cue:201]: Good eye!

**Review by lucperkins — APPROVED** [review]

File: website/cue/reference/cli.cue (hunk 1 above)
Comment (DavidWeid) [reply website/cue/reference/cli.cue:201]: Thanks!

**Comment by DavidWeid** [comment]
@lucperkins Thanks for pointing out the colors of li:before. This has been standardized now along with `li` text itself. Also saw an issue with code/link elements nested under `li`'s that was affecting `prose-sm`

**Comment by lucperkins** [comment]
Ship it!

**Pull request merged**